### PR TITLE
Superstep java

### PR DESCRIPTION
**Pull request by tkutz**

### CODE REVIEW

File: plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend [hunk 1]
```diff
@@ -28,28 +29,45 @@ class RunCycle implements org.yakindu.sct.generator.core.submodules.lifecycle.Ru
...
+			switch (stateVector[nextStateIndex]) {
+			«FOR state : flow.states.filter[isLeaf]»
+				«IF state.reactMethod !== null»
+					case «state.stateName.asEscapedIdentifier»:
+						«state.reactMethod.shortName»(true);
+						break;
+			«ENDIF»
+			«ENDFOR»
+			default:
+				// «getNullStateName()»
```
Comment (rherrmannr) [inline plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend]: remove comment

File: plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend [hunk 2]
```diff
@@ -28,28 +29,45 @@ class RunCycle implements org.yakindu.sct.generator.core.submodules.lifecycle.Ru
...
 	
+	def protected runCycleForLoop(ExecutionFlow flow) '''
+		for (nextStateIndex = 0; nextStateIndex < stateVector.length; nextStateIndex++) {
+			switch (stateVector[nextStateIndex]) {
+			«FOR state : flow.states.filter[isLeaf]»
+				«IF state.reactMethod !== null»
+					case «state.stateName.asEscapedIdentifier»:
+						«state.reactMethod.shortName»(true);
+						break;
+			«ENDIF»
```
Comment (rherrmannr) [inline plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend]: formatting

File: plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend (hunk 1 above)
Comment (tkutz) [reply plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend]: this is intended

File: plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend (hunk 2 above)
Comment (tkutz) [reply plugins/org.yakindu.sct.generator.java/src/org/yakindu/sct/generator/java/submodules/lifecycle/RunCycle.xtend]: done

**Pull request merged**